Foundation/FileManager+POSIX.swift

```diff
@@ -947,12 +953,16 @@ extension FileManager {
                 }
                 return (statInfo, creationDate)
             } else {
-                // fallback if statx() is unavailable or fails
-                let statInfo = try _lstatFile(atPath: path, withFileSystemRepresentation: fsRep)
-                return (statInfo, nil)
+                return try _statxFallback(atPath: path, withFileSystemRepresentation: fsRep)
             }
         }
     }
+
+    // Fallback if statx() is unavailable or fails
+    private func _statxFallback(atPath path: String, withFileSystemRepresentation fsRep: UnsafePointer<Int8>?) throws -> (stat, Date?) {
+        let statInfo = try _lstatFile(atPath: path, withFileSystemRepresentation: fsRep)
+        return (statInfo, nil)
+    }
```

Comment:
I would move this function to be inside `_statxFile(atPath:)` since that is the only place it is called from.